apps/meteor/server/models/raw/Users.js

```diff
@@ -270,18 +262,115 @@ export class UsersRaw extends BaseRaw {
 		return this.findPaginated(query, options);
 	}
 
+	findPaginatedActiveUsersByRoomIdWithHighestRole(
+		searchTerm,
+		rid,
+		searchFields,
+		options = {},
+		extraQuery = [],
+		{ startsWith = false, endsWith = false } = {},
+	) {
+		const regexString = (startsWith ? '^' : '') + escapeRegExp(searchTerm) + (endsWith ? '$' : '');
+		const termRegex = new RegExp(regexString, 'i');
+
+		const orStmt = (searchFields || []).map((el) => ({ [el.trim()]: termRegex }));
+
+		const query = {
+			$and: [
+				{
+					active: true,
+					__rooms: rid,
+					username: { $exists: true },
+					// if the search term is empty, don't need to have the $or statement (because it would be an empty regex)
+					...(searchTerm && orStmt.length && { $or: orStmt }),
+				},
+				...extraQuery,
+			],
+		};
+
+		const skip =
+			options.skip !== 0
+				? [
+						{
+							$skip: options.skip,
+						},
+				  ]
+				: [];
+
+		const limit =
+			options.limit !== 0
+				? [
+						{
+							$limit: options.limit,
+						},
+				  ]
+				: [];
+
+		return this.col.aggregate([
```

Comment:
A suggestion about this aggregation: i don't see any index that could cover the whole match, it will go either on `active` idx or `__rooms` idx, depending on which one is more selective. 

Can we try to run an explain on an env with a ton of users? Like open, just to see how it performs? It touches 2 of the biggest collections so this could take a ton of time to compute 😬

Reply:
Hmmm that'd be nice. I'll talk to the architecture team and check if that's possible
Anyways, my guess here is that it should perform fine :thinking: it's essentially the same query we use in `findPaginatedByActiveUsersExcept` (the `__rooms` condition is not explicitly stated there, but it is defined in the `extraQuery` param), which is used in the `.members` endpoints currently (within the `findUsersOfRoom` method)
I can't spot any difference between them other than this new one is running in an aggregation
cc @sampaiodiego

Reply:
Please consider this while reviewing, @ggazzo @sampaiodiego